src/Mod/Test/BaseTests.py

```diff
@@ -253,6 +253,22 @@ def testRotAndMoveMatrix(self):
         self.assertEqual(m2, m3*m4   , "Wrong multiplication order")
         self.assertNotEqual(m2, m4*m3, "Wrong multiplication order")
 
+
+    def testRotAndScaleMatrix(self):
+        m1 = FreeCAD.Matrix()
+        m1.rotateZ(.2)
+        m1.scale(0.5)
+        m1.rotateY(.2)
+        m1.scale(3)
+        m1.move(10,5,-3)

```

Comment:
Is the move a necessary component of the test?

Reply:
There where cases I found during developing this fix where the move affected the output. The move is part of the matrix that is passed to the function so it surely could be the source for a future bug.

Reply:
Now as we have set up the basic infrastructure for real unit tests there should also be written some tests in C++. See tests/src/Base/.